chore: [IOBP-123] New list item transaction into IDPay timeline

Hantex9 opened the pull request:
# ⚠️ This PR depends on #4776 ⚠️

## Short description
This PR adds into IDPay timeline operations the new `ListItemTransaction` component;

## List of changes proposed in this pull request
- `TimelineOperationsListItem.tsx`: Removed the previous ad-hoc component implementation with the new component into DS;
- Adapted business logic to calculate the title, amount, icon and status required in the `ListItemTransaction` component;

## How to test
Into wallet section, open any IDPay initiative and check that the new component is showed correctly into timeline operation list

## Preview

https://github.com/pagopa/io-app/assets/34343582/f53e4c5d-fdec-4af3-ba10-3261ac428d83

mastro993 commented:
Looks good! Just a little observation before merging.

The design on Figma has new icons for transactions and onboarding items:

<img src="https://github.com/pagopa/io-app/assets/6160324/3a97724b-7d3e-4072-810d-d8485eeda857" width="200" />

What do you think about replacing the current ones in this PR?

Hantex9 commented:
> Looks good! Just a little observation before merging.
> 
> The design on Figma has new icons for transactions and onboarding items:
> What do you think about replacing the current ones in this PR? Also, seems like the Divider between items is missing.

Yep, absolutely! 
Addressed in 73c2f11cd56526cd4829b070347dcf5418bb4b33

In the commit d758f7b35a1e251a215df87dca9a312dff66d8b1 I have also changed the "Ultimo aggiornamento" label date format in order to be design compliant

## Preview
<img src="https://github.com/pagopa/io-app/assets/34343582/8c598a6f-d4a5-4ccc-8448-f354bea884d8" width="250" />

forrest57 commented on ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:
```diff
@@ -61,78 +54,45 @@ const OperationIcon = ({ operation }: OperationComponentProps) => {
-    : {};
-
   const getOperationTitle = () => {
     switch (operation.operationType) {
```
```suggestion
    switch (operation.operationType) {
case InstrumentOperationTypeEnum.ADD_INSTRUMENT:
        return I18n.t(
          "idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.ADD_INSTRUMENT",
          {
            maskedPan: operation.maskedPan
          }
        );
```
to avoid missing I18n errors, we need to specify it, it probably went lost in all the changes 👍 
<img width="291" alt="Screenshot 2023-07-13 at 17 03 58" src="https://github.com/pagopa/io-app/assets/60693085/c8788e01-57ed-4eb9-a7ea-ecf74f55c76d">

forrest57 replied:
that said, seeing the designs the maskedPan is probably getting removed, so I'd ask @thisisjp about this

mastro993 commented:
> > Looks good! Just a little observation before merging.
> > The design on Figma has new icons for transactions and onboarding items:
> > What do you think about replacing the current ones in this PR? Also, seems like the Divider between items is missing.
> 
> Yep, absolutely! Addressed in [73c2f11](https://github.com/pagopa/io-app/commit/73c2f11cd56526cd4829b070347dcf5418bb4b33)
> 
> In the commit [d758f7b](https://github.com/pagopa/io-app/commit/d758f7b35a1e251a215df87dca9a312dff66d8b1) I have also changed the "Ultimo aggiornamento" label date format in order to be design compliant
> 
> PS: Dividers are present but the video was made with a downscaled simulator and they are not visible
> 
> ## Preview
> <img alt="" width="250" src="https://user-images.githubusercontent.com/34343582/253294451-8c598a6f-d4a5-4ccc-8448-f354bea884d8.png">

Are you sure about dividers? In previous items were visible.

Hantex9 commented:
## Update
In the commit b350a4bbe664e8ef555277f210a2f028964e9bba has been refactored the ListItemTransaction behavior into `TimelineOperationListItem`

## Preview

|Before|After|
|-|-|
| <img src="https://github.com/pagopa/io-app/assets/34343582/eb38895e-cf61-410a-96ca-1be00f49d22c" width="250" /> | <img src="https://github.com/pagopa/io-app/assets/34343582/70028532-0bb2-4f86-b270-9a03c8211c70" width="250" /> |

@thisisjp

mastro993 reviewed (approved):
LGTM! Regarding the divider that is not visible, I checked on a real device and seems to be ok! 👌🏻
